spring-pulsar/src/test/java/org/springframework/pulsar/listener/reactive/ReactivePulsarListenerTests.java

```diff
@@ -610,4 +610,64 @@ ReactiveMessageConsumerBuilderCustomizer<?> subscriptionInitialPositionEarliest(
 
 	}
 
+	@Nested
+	@ContextConfiguration(classes = PulsarListenerConcurrencyTestCases.TestPulsarListenersForConcurrency.class)
+	class PulsarListenerConcurrencyTestCases {
+
+		static CountDownLatch latch = new CountDownLatch(100);
+
+		static BlockingQueue<String> queue = new LinkedBlockingQueue<>();
+
+		@Test
+		void pulsarListenerWithConcurrency() throws Exception {
+			for (int i = 0; i < 100; i++) {
+				pulsarTemplate.send("pulsarListenerConcurrency", "hello foo");
+			}
+			assertThat(latch.await(5, TimeUnit.SECONDS)).isTrue();
+		}
+
+		@Test
+		void pulsarListenerWithConcurrencyKeyOrdered() throws Exception {
+			pulsarTemplate.newMessage("first").withTopic("pulsarListenerWithConcurrencyKeyOrdered")
+					.withMessageCustomizer(m -> m.key("key")).send();
+			pulsarTemplate.newMessage("second").withTopic("pulsarListenerWithConcurrencyKeyOrdered")
+					.withMessageCustomizer(m -> m.key("key")).send();
+			assertThat(queue.poll(5, TimeUnit.SECONDS)).isEqualTo("first");
+			assertThat(queue.poll(5, TimeUnit.SECONDS)).isEqualTo("second");
+		}
+
+		@EnablePulsar
+		@Configuration
+		static class TestPulsarListenersForConcurrency {
+
+			@ReactivePulsarListener(topics = "pulsarListenerConcurrency", consumerCustomizer = "consumerCustomizer",
+					concurrency = "100")
+			Mono<Void> listen1(String message) {
+				latch.countDown();
+				// if messages are not handled concurrently, this will make the latch
+				// await timeout.
+				return Mono.delay(Duration.ofMillis(100)).then();
```

Comment:
Clever way to test the concurrency.

Reply:
Credits goes to @lhotari in the reactive client tests from which I shamelessly borrowed the inspiration 😃

Reply:
Well, now I will shamelessly borrow from you as well. PR reviews are one of the greatest teaching/learning vessels. 👍🏻